### PR TITLE
Updated build pipeline

### PR DESCRIPTION
**Pull request by Knyatsine**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,36 +1,30 @@
+
```
Comment (jack-cloud-platform) [inline azure-pipelines.yml:1]:
```suggestion
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -1,36 +1,30 @@
...
 
 pool:
-  name: DAS - Continuous Integration
   workspace:
     clean: all
+  name: 'DAS - Continuous Integration'
 
 resources:
   repositories:
-  - repository: self
```
Comment (jack-cloud-platform) [inline azure-pipelines.yml:22]: If the self reference isn't needed, good if you can PR its removal from  the das-github-template https://github.com/SkillsFundingAgency/das-github-template/blob/main/azure-pipelines.yml#L22

**Comment by jack-cloud-platform** [comment]
We've changed to unquoted string values in https://github.com/SkillsFundingAgency/das-github-template/pull/3

File: pipeline-templates/job/code-build.yml [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+  pool:
+    name: DAS - Continuous Integration
   variables:
   - group: BUILD Management Resources
+  - name: SolutionBaseName
+    value: 'SFA.DAS.Roatp.Functions' 
+
   steps:
-    - task: gitversion/setup@0
+    - task: gittools.gittools.setup-gitversion-task.gitversion/setup@0
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/code-build.yml:11]: The task can remain as the shortened gitversopm/setup@0 as in the das-github-tempate

File: pipeline-templates/job/code-build.yml [hunk 4]
```diff
@@ -23,25 +24,25 @@ jobs:
...
     - task: DotNetCoreCLI@2
-      displayName: Publish - dotnet publish ${{ SolutionBaseName }}
+      displayName: "Publish $(SolutionBaseName)"
       inputs:
         command: publish
         publishWebProjects: false #set to false as this setting (which defaults to true) will parse the entire repo for web projects
-        projects: src/${{ SolutionBaseName }}/${{ SolutionBaseName }}.csproj
-        arguments: -o $(build.artifactstagingdirectory)/publish -c ${{ BuildConfiguration }} --no-build
+        projects: src/$(SolutionBaseName)/$(SolutionBaseName).csproj
+        arguments: "-o $(build.artifactstagingdirectory)/publish -c $(BuildConfiguration) --no-build"
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/code-build.yml:32]: das-github-template has BuildConfiguration passed as a parameter from azure-pipelines.yml for explicitness

File: pipeline-templates/job/code-build.yml [hunk 5]
```diff
@@ -23,25 +24,25 @@ jobs:
...
 
     - task: DotNetCoreCLI@2
-      displayName: Publish - dotnet publish ${{ SolutionBaseName }}
+      displayName: "Publish $(SolutionBaseName)"
       inputs:
         command: publish
         publishWebProjects: false #set to false as this setting (which defaults to true) will parse the entire repo for web projects
-        projects: src/${{ SolutionBaseName }}/${{ SolutionBaseName }}.csproj
-        arguments: -o $(build.artifactstagingdirectory)/publish -c ${{ BuildConfiguration }} --no-build
+        projects: src/$(SolutionBaseName)/$(SolutionBaseName).csproj
```
Comment (jack-cloud-platform) [inline pipeline-templates/job/code-build.yml:31]: das-github-template has SolutionBaseName passed as a parameter from azure-pipelines.yml for explicitness. If conventions should change, there should be a PR to das-github-template.

File: pipeline-templates/job/code-build.yml (hunk 4 above)
Comment (Knyatsine) [reply pipeline-templates/job/code-build.yml:32]: @jack-cloud-platform so do i need to remove it ?

Comment (jack-cloud-platform) [reply pipeline-templates/job/code-build.yml:32]: das-github-template standards should ideally not be moved away from. You can propose changes to das-github-template though

Comment (Knyatsine) [reply pipeline-templates/job/code-build.yml:32]: @jack-cloud-platform No worries, i will check it out tomorrow. Thanks for the review

File: pipeline-templates/job/code-build.yml [hunk 6]
```diff
@@ -23,13 +26,13 @@ jobs:
     - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks
 
     - task: DotNetCoreCLI@2
-      displayName: Publish - dotnet publish ${{ SolutionBaseName }}
+      displayName: Publish $(SolutionBaseName)
```
Comment (nbowes24) [inline pipeline-templates/job/code-build.yml:29]: This was removed from the parameters along with $BuildConfiguration

File: pipeline-templates/job/code-build.yml [hunk 7]
```diff
@@ -4,9 +4,12 @@ parameters:
 
 jobs:
 - job: CodeBuild
+  pool:
+    name: DAS - Continuous Integration
```
Comment (nbowes24) [inline pipeline-templates/job/code-build.yml:8]: This is already defined in azure-pipelines.yml

**Review by nbowes24 — APPROVED** [review]

**Pull request merged**